python/simple/v1/app.py

```diff
@@ -24,60 +24,90 @@
 ampli.track(None, EventNoProperties(), event_options=EventOptions(user_id="user_id", device_id="device_id"))
 
 # track event with strongly typed method
-ampli.event_no_properties("user_id", event_options=EventOptions(device_id="device_id"))
+ampli.event_no_properties("user_id", EventNoProperties(), event_options=EventOptions(device_id="device_id"))
 
 # track EventWithAllProperties
-ampli.event_with_all_properties(user_id="user_id",
-                                required_array=["a", "b"],
-                                required_boolean=False,
-                                required_enum=EventWithAllProperties.RequiredEnum.ENUM_1,
-                                required_number=16.4,
-                                required_integer=3,
-                                required_string="str")
-ampli.track("user_id", EventWithAllProperties(required_array=["a", "b"],
-                                              required_boolean=False,
-                                              required_enum=EventWithAllProperties.RequiredEnum.ENUM_1,
-                                              required_number=16.4,
-                                              required_integer=3,
-                                              required_string="str"))
+ampli.event_with_all_properties(
+    "user_id",
+    EventWithAllProperties(
+        required_array=["a", "b"],
+        required_boolean=False,
+        required_enum=EventWithAllProperties.RequiredEnum.ENUM_1,
+        required_number=16.4,
+        required_integer=3,
+        required_string="str"
+    ))
+
+ampli.track(
+    "user_id",
+    EventWithAllProperties(
+        required_array=["a", "b"],
+        required_boolean=True,
+        required_enum=EventWithAllProperties.RequiredEnum.ENUM_2,
+        required_number=16.4,
+        required_integer=3,
+        required_string="str"
+    ))
 
 # track event with const properties
-ampli.event_with_const_types(user_id="user_id")
+ampli.event_with_const_types(user_id="user_id", event=EventWithConstTypes())
```

Comment:
nit: maybe remove prop names (user_id, event) and use positional to make the example more concise. e.g. `ampli.event_with_const_types("user_id",EventWithConstTypes())`

Reply:
@justin-fiedler In Jre/Kotlin if an event has no properties or has only constant properties:
```
fun eventWithConstTypes(userId: String?, options: EventOptions? = null, extra: MiddlewareExtra? = null) {
    this.track(userId, EventWithConstTypes(), options, extra)
}

fun eventNoProperties(userId: String?, options: EventOptions? = null, extra: MiddlewareExtra? = null) {
    this.track(userId, EventNoProperties(), options, extra)
}
```

Reply:
I see. Should I make event with no parameter optional here too? @justin-fiedler

Reply:
@bohan-amplitude Ideally event with no properties created methods without the `event` param.

fun eventWithConstTypes(userId: String?, options: EventOptions? = null) {
    this.track(userId, EventWithConstTypes(), options)
}